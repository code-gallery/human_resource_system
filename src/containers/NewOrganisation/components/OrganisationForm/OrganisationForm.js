import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const OrganisationForm = (props) => {
  const { name, onChangeInput, save, errors, success } = props

  return (
    <div>
      <h2 className="form-group">Tell us about an organisation</h2>
      <p className="form-group">
        APPII team will use the information below to onboard an organisation on our system.<br />
        The fields with * are mandatory.
      </p>
      <form>
        <div
          className={classNames('form-group', {
            'has-error': errors.indexOf('name') !== -1
          })}
        >
          <label>Organisation name *</label>
          <input
            name="name"
            type="text"
            onChange={onChangeInput}
          />
        </div>

        <div
          className={classNames('form-group', {
            'has-error': errors.indexOf('contactName') !== -1
          })}
        >
          <label>Contact name *</label>
          <input
            name="contactName"
            type="text"
            onChange={onChangeInput}
          />
        </div>

        <div
          className={classNames('form-group', {
            'has-error': errors.indexOf('contactEmail') !== -1
          })}
        >
          <label>Contact email *</label>
          <input
            name="contactEmail"
            type="text"
            onChange={onChangeInput}
          />
        </div>

        <div
          className={classNames('form-group', {
            'has-error': errors.indexOf('websiteUrl') !== -1
          })}
        >
          <label>Website url *</label>
          <input
            name="websiteUrl"
            type="text"
            onChange={onChangeInput}
          />
        </div>

        <div className="form-group">
          <label>LinkedIn profile url</label>
          <input
            name="linkedInUrl"
            type="text"
            onChange={onChangeInput}
          />
        </div>

        <div className="form-group">
          <button
            className="form-btn save"
            onClick={save}
          >
            Send
          </button>
        </div>
      </form>
      {success &&
        <p className="form-group">
          Thanks, we will create a profile for {name} and seek a verification administrator as quickly as we can
        </p>
      }
    </div>
  )
}

OrganisationForm.propTypes = {
  onChangeInput: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
  errors: PropTypes.array.isRequired,
  success: PropTypes.bool.isRequired,
  name: PropTypes.string
}

export default OrganisationForm
