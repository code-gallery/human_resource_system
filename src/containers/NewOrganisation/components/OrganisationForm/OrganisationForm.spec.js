import React from 'react'
import renderer from 'react-test-renderer'
import OrganisationForm from './OrganisationForm'

describe('containers/NewOrganisation/components/OrganisationForm', () => {
  const commonProps = {
    save: jest.fn(),
    onChangeInput: jest.fn(),
    success: false
  }

  describe('when there no errors', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        errors: []
      }
      const tree = renderer.create(
        <OrganisationForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when name and websiteUrl have errors', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        errors: [ 'name', 'websiteUrl' ]
      }
      const tree = renderer.create(
        <OrganisationForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when success is true', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        errors: [],
        name: 'Applied',
        success: true
      }
      const tree = renderer.create(
        <OrganisationForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
