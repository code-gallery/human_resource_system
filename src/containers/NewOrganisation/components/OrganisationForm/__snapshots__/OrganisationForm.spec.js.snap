// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/NewOrganisation/components/OrganisationForm when name and websiteUrl have errors renders correctly 1`] = `
<div>
  <h2
    className="form-group"
  >
    Tell us about an organisation
  </h2>
  <p
    className="form-group"
  >
    APPII team will use the information below to onboard an organisation on our system.
    <br />
    The fields with * are mandatory.
  </p>
  <form>
    <div
      className="form-group has-error"
    >
      <label>
        Organisation name *
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact name *
      </label>
      <input
        name="contactName"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact email *
      </label>
      <input
        name="contactEmail"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group has-error"
    >
      <label>
        Website url *
      </label>
      <input
        name="websiteUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        LinkedIn profile url
      </label>
      <input
        name="linkedInUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="form-btn save"
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </form>
</div>
`;

exports[`containers/NewOrganisation/components/OrganisationForm when success is true renders correctly 1`] = `
<div>
  <h2
    className="form-group"
  >
    Tell us about an organisation
  </h2>
  <p
    className="form-group"
  >
    APPII team will use the information below to onboard an organisation on our system.
    <br />
    The fields with * are mandatory.
  </p>
  <form>
    <div
      className="form-group"
    >
      <label>
        Organisation name *
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact name *
      </label>
      <input
        name="contactName"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact email *
      </label>
      <input
        name="contactEmail"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Website url *
      </label>
      <input
        name="websiteUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        LinkedIn profile url
      </label>
      <input
        name="linkedInUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="form-btn save"
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </form>
  <p
    className="form-group"
  >
    Thanks, we will create a profile for 
    Applied
     and seek a verification administrator as quickly as we can
  </p>
</div>
`;

exports[`containers/NewOrganisation/components/OrganisationForm when there no errors renders correctly 1`] = `
<div>
  <h2
    className="form-group"
  >
    Tell us about an organisation
  </h2>
  <p
    className="form-group"
  >
    APPII team will use the information below to onboard an organisation on our system.
    <br />
    The fields with * are mandatory.
  </p>
  <form>
    <div
      className="form-group"
    >
      <label>
        Organisation name *
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact name *
      </label>
      <input
        name="contactName"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Contact email *
      </label>
      <input
        name="contactEmail"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Website url *
      </label>
      <input
        name="websiteUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        LinkedIn profile url
      </label>
      <input
        name="linkedInUrl"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="form-btn save"
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </form>
</div>
`;
